src/btree/row_modify.c

```diff
@@ -409,12 +409,9 @@ __wt_update_obsolete_check(
          * Decrement the dirty byte count while holding the page lock, else we can race with
          * checkpoints cleaning a page.
          */
-        if (update_accounting) {
-            for (size = 0, upd = next; upd != NULL; upd = upd->next)
-                size += WT_UPDATE_MEMSIZE(upd);
-            if (size != 0)
-                __wt_cache_page_inmem_decr(session, page, size);
-        }
+        for (size = 0, upd = next; upd != NULL; upd = upd->next)
+            size += WT_UPDATE_MEMSIZE(upd);
+        __wt_cache_page_inmem_decr(session, page, size);
```

Comment:
Did you check for safety and decide that the `size != 0` check that was here is wasted work?

Reply:
There is at least one `upd` that exists for free. We already checked for NULL before we entered the function. So the `size` can never be 0. I can add an assert to check it to confirm it.

Reply:
I see - thanks. I *think* there are cases where update structures have a zero size, I vaguely recall @marcbutler running into that a while ago when making changes to the size-checking code around them.

Reply:
Yes, it is possible that the update value is zero. The `WT_UPDATE_MEMSIZE` includes the `WT_UPDATE` structure size as well. So it cannot be zero.

Reply:
Yes there is `WT_UPDATE_SIZE_NOVALUE` > `WT_UPDATE_SIZE` (they differ by a byte).

I'm not sure it is relevant here here but `WT_UPDATE_MEMSIZE` is an *overestimate* of memory used by the update. So: `WT_UPDATE_MEMSIZE` == 64 if `upd->size == 0`. I'm not sure how if at all this would affect accounting. I'm assuming it doesn't matter if it is consistent.

> There is at least one `upd` that exists for free. We already checked for NULL before we entered the function. So the `size` can never be 0. I can add an assert to check it to confirm it.

I'm curious -- why not check within the function?